src/mempool/README.md

```diff
@@ -0,0 +1,70 @@
+layer). The validity of transactions is defined by the Ledger layer. Notice a
+change in the selected chain should trigger a prompt revalidation of the pending
+transactions to discard those that became invalid on the new selection.  Such
+collection of valid transactions will be requested by the Consensus layer to
+[forge new blocks](../consensus/forging.md).
+
+Being valid implies in particular that the transaction is pending, i.e. it has
+not been included in any prior block ever, as in Cardano every valid transaction
+must consume at least one UTxO.
+
+> [!NOTE]
+>
+> Notice that a transaction might be considered valid in one peer, but invalid
+> in another peer which has selected a different chain. Even in the case both
+> peers have the same selection, the existence in the mempool (or lack thereof)
+> of some other specific transaction that creates (or consumes) the inputs for
+> this one could lead to different veredicts on each peer.
+
+While only nodes that forge new blocks will use these transactions to create
+blocks, non-block-producing nodes are still expected to diffuse transactions to
+their peers. **All participating nodes** must be able to receive, validate, and
+distribute transactions.
+
+> [!TIP]
+>
+> When a node is shut down, the implementation can choose what to do with the
+> pending transactions. The reference Haskell implementation discards them.
+> However, it is conceivable for a node to store on disk the pending
+> transactions from its mempool.
+
+The [`TxSubmission2`](./txsubmission2.md) miniprotocol is the
+one used to diffuse the pending transactions through the network. Any mempool
+implementation has some requirements to allow diffusing transactions via
+`TxSubmission2` and to fulfill the requirements of the Consensus layer in terms
+of forging blocks:
+- Acquiring a snapshot of valid transactions for a newly forged block should be
+  as fast as possible, as it will delay all other steps in the diffusion of such
+  a block.
+- Cursor-like access to pending transactions, to fulfill the `TxSubmission2`
+  protocol requirements.
+- Re-validation on a new selected chain: it is useless to keep transactions that
+  are no longer valid in the mempool.
+
+Notice that _what_ a transaction actually is defined by the Ledger layer and it
+depends on the actual era in action on the tip of the chain. The Ledger layer
+provides mechanisms to translate transactions from older eras to more recent
+ones. Once a transaction is translated to a later era, it is such translation
+the one that is distributed to the network afterwards.
+
+In order to prevent overusing resources, the Ledger places a limit on the number
+of pending transactions it accepts. Once this limit is reached, requests to add
+new transactions will be rejected. This _back-pressure_ mechanism is critically
+important in periods of intense traffic to preserve the overall throughput of
+the network.
+
```

Comment:
I made a note to write about fairness in the mempool, and what the ref impl currently does.

Reply:
I think fairness is not something that involves us or the _abstract_ consensus or mempool. It is a nice to have but not critical for the network, it is however critical for users.

Reply:
Should other implementations of the mempool also exhibit this level of fairness? If the answer is yes, it probably is a good idea to write about it here :)

Reply:
Added a TODO